### ISSUE TITLE
Support for unicode email attachments filenames

### ISSUE THREAD

**Issue by punxrok** [jstedfast/MimeKit#231]
Hello.

I'm having problems with correctly displaying unicode email attachments filenames in Outlook.
Would it be possible to extend MimeKit in the way that saved eml files would be  compatible with user agents which do not implement RFC 5987 (see https://tools.ietf.org/html/rfc6266 chapter 5)?

Currenty MimeKit does that:
Content-Type: image/jpg; name*=utf-8''%E2%82%AC%20money.jpg
Content-Disposition: attachment; filename*=utf-8''%E2%82%AC%20money.jpg
Content-Transfer-Encoding: base64

With modification that correctly displays attachment filename in Outlook:
Content-Type: image/jpg; name*=utf-8''%E2%82%AC%20money.jpg; name="=?utf-8?B?4oKsIG1vbmV5LmpwZw==?="
Content-Disposition: attachment; filename*=utf-8''%E2%82%AC%20money.jpg
Content-Transfer-Encoding: base64

Example program is available [here](http://s000.tinyupload.com/index.php?file_id=42515934625450320075).

Best regards,
Rok

**Comment by jstedfast** [comment]
rfc6266 is for HTTP and does not apply to e-mail.

That said, however, you are in luck :-)

I actually implemented support for sending name="=?utf-8?B?...?=" style attachment file names a few months ago.

I need to update `FAQ.md` to have this information because this feature is not immediately obvious (and `FAQ.md` is probably where most people look first, expecting ti to be as up-to-date as the mimekit.org docs), but I did write up a FAQ about it on the docs website:

http://www.mimekit.org/docs/html/FrequentlyAskedQuestions.htm#UntitledAttachments

Let me know if you have any questions!
